### PR TITLE
Enahncement/1356 lint packages upgrade

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #1356 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: tests/e2e/matchers/to-have-tracking.js [hunk 1]
```diff
@@ -20,7 +20,6 @@
  * External dependencies
  */
 import { getDefaultOptions } from 'expect-puppeteer';
-import { Page, ElementHandle } from 'puppeteer';
```
Comment (felixarntz) [inline tests/e2e/matchers/to-have-tracking.js:23]: Why was this removed here (also in the other file)? Shouldn't we import them so that the reference in the docs below is interpreted correctly?

File: .eslintrc.json [hunk 2]
```diff
@@ -11,6 +11,8 @@
     "fetchMock": false
   },
   "rules": {
+    "jsdoc/no-undefined-types": "off",
+    "jsdoc/require-param": "off",
```
Comment (felixarntz) [inline .eslintrc.json:15]: These two rules shouldn't be turned off, instead the violations should be fixed by importing the relevant types / adding the necessary param docs.

File: .stylelintrc [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "at-rule-no-unknown": null,
     "declaration-property-unit-whitelist": null,
     "function-url-quotes": "always",
+    "selector-id-pattern": null,
```
Comment (felixarntz) [inline .stylelintrc:13]: Same here, let's not globally ignore these types of issues because they should get flagged. In this case, all problems seem to be caused by `#google_dashboard_widget` which however is generated by WordPress. So let's instead annotate these specific lines to be ignore by stylelint.

File: tests/e2e/matchers/to-have-tracking.js (hunk 1 above)
Comment (eugene-manuilov) [reply tests/e2e/matchers/to-have-tracking.js:23]: > Why was this removed here (also in the other file)?

Eslint blames us that these are unused vars. So, i have decided to remove it (since technically speaking - it's not needed there) instead of disabling eslint check for that line.

> Shouldn't we import them so that the reference in the docs below is interpreted correctly?

Do we maintain publicly available documentation for this? Do we really need it to be interpreted correctly?

Comment (eugene-manuilov) [reply tests/e2e/matchers/to-have-tracking.js]: Done.

File: .stylelintrc (hunk 3 above)
Comment (eugene-manuilov) [reply .stylelintrc]: Ok, updated.

File: .eslintrc.json (hunk 2 above)
Comment (eugene-manuilov) [reply .eslintrc.json]: Ok, updated.
